Heat

YurgenJurgensen opened the pull request:
This builds and appears to run.

YurgenJurgensen committed:
First pass of very WIP grinder.

YurgenJurgensen committed:
Merge remote-tracking branch 'origin/kt1.9' into kt1.9

# Conflicts:
#	src/main/kotlin/com/cout970/magneticraft/registry/Blocks.kt

YurgenJurgensen committed:
Merge remote-tracking branch 'origin/kt1.9' into kt1.9

YurgenJurgensen committed:
Some grinder fixes.  Wood and coal processing.

YurgenJurgensen committed:
Some grinder fixes.  Wood and coal processing.

YurgenJurgensen committed:
Grinder now has secondary outputs and recipe.

YurgenJurgensen committed:
Grinder now has secondary outputs and recipe.

YurgenJurgensen committed:
WIP

YurgenJurgensen committed:
Slightly Less WIP

YurgenJurgensen committed:
Slightly Less WIP

YurgenJurgensen committed:
Kind-of Working

YurgenJurgensen committed:
Fix electric heater

YurgenJurgensen committed:
Fix max fuel temp

YurgenJurgensen committed:
Refactored some things.  Now heat blocks emit light.

YurgenJurgensen committed:
Merge remote-tracking branch 'origin/kt1.9' into firebox

YurgenJurgensen committed:
Pipe rendering now working with blockstates, credit to DrCyano for the horrible json monstrosity that is the placeholder model files.

YurgenJurgensen committed:
Machine overheating first pass

YurgenJurgensen committed:
WIP Icebox

YurgenJurgensen committed:
Less WIP Icebox

YurgenJurgensen committed:
Fixed recipe bug

YurgenJurgensen committed:
Added freezing

YurgenJurgensen committed:
Improved ambient temperature handling

YurgenJurgensen committed:
Minor list optimisation

YurgenJurgensen committed:
Heat working with multiblocks?

YurgenJurgensen committed:
Kiln Intermediate Products

YurgenJurgensen committed:
Kiln damages entities, has bounding box

YurgenJurgensen committed:
Kiln damages entities, has bounding box

YurgenJurgensen committed:
Merged in BlockFallingBase

YurgenJurgensen committed:
Kiln now smelts.  Only a few graphical bugs to fix now.

YurgenJurgensen committed:
Now with particles and fewer bounding box errors

YurgenJurgensen committed:
Solar Panels are more effective in hot biomes

YurgenJurgensen committed:
First pass at JEI

YurgenJurgensen committed:
Redstone-switchable heat pipes

YurgenJurgensen committed:
Stamped and heavy duty items, basic

YurgenJurgensen committed:
Incendiary generator supports max fuel heat

YurgenJurgensen committed:
Fuel temperature tooltips.

YurgenJurgensen committed:
Grinder model, damage types

YurgenJurgensen committed:
Mesh recipe

YurgenJurgensen committed:
Language/Texture fixes

YurgenJurgensen committed:
Model translation fix

YurgenJurgensen committed:
Grinder updated to new IO system

YurgenJurgensen committed:
Skeleton classes for sifter

YurgenJurgensen committed:
Some implementation/fixes for sifter

YurgenJurgensen committed:
Sifter now kind-of sifts.  Is still invisible.

cout970 committed:
Fixed formatting issues
Moved head tileEntities to head package
Changed HeatContainer to be TileEntity independent, will be needed for multiparts
Optimized Imports
Removed recipe for BlockThermometer (issues with jei?)
Changed fuelTempHelper by FuelCache
Moved GlStateManager calls to TileEntityRenderer
Updated beehive_kiln model adding normals to avoid weird crashes //TODO fix the texture
Updated kotlin to 1.1-M04

cout970 committed:
Changed  biomeTempToKelvin
Changed some !worldObj.isRemote to worldObj.isServer
Moved DameSources to a separated file
Added some placeholder textures

cout970 committed:
Updated kiln model, still not finished

cout970 committed:
Updated textures
Fixed blocks not being registered correctly

cout970 committed:
Fixed TileInfiniteHeat, it didn't have a empty constructor
Minor comment/code formatting issues

YurgenJurgensen committed:
Merge?

YurgenJurgensen committed:
Merge?

cout970 commented on src/main/kotlin/com/cout970/magneticraft/tileentity/heat/TileBrickFurnace.kt:
```diff
@@ -22,8 +22,8 @@ class TileBrickFurnace : TileHeatBase() {
-            worldGetter = this::getWorld,
-            posGetter = this::getPos)
+            worldGetter = { this.world },
+            posGetter = { this.getPos() })
```
You need to refresh the gradle project to download the lastest version of kotlin and be able to use 'this::getWorld' instead of '{ this.world }'

cout970 commented on src/main/kotlin/com/cout970/magneticraft/api/internal/heat/HeatConnection.kt:
```diff
@@ -0,0 +1,43 @@
+ */
+open class HeatConnection(
+        private var firstNode: IHeatNode,
+        private var secondNode: IHeatNode
```
This two properties must be val instead of var

cout970 commented on src/main/kotlin/com/cout970/magneticraft/api/internal/registries/machines/heatExchanger/HeatExchangerRecipe.kt:
```diff
@@ -0,0 +1,31 @@
+package com.cout970.magneticraft.api.internal.registries.machines.tablesieve
```
Pls update the package

cout970 commented on src/main/kotlin/com/cout970/magneticraft/api/internal/registries/machines/heatExchanger/HeatExchangerRecipeManager.kt:
```diff
@@ -0,0 +1,40 @@
+package com.cout970.magneticraft.api.internal.registries.machines.tablesieve
```
same, update the package

cout970 commented on src/main/kotlin/com/cout970/magneticraft/api/internal/registries/machines/heatExchanger/IceboxRecipe.kt:
```diff
@@ -0,0 +1,38 @@
+package com.cout970.magneticraft.api.internal.registries.machines.tablesieve
```
this should be in a different folder, icebox instead of heatExchanger, also the package heatExchanger should be lowercase

cout970 commented on src/main/kotlin/com/cout970/magneticraft/api/internal/registries/machines/heatExchanger/IceboxRecipeManager.kt:
```diff
@@ -0,0 +1,57 @@
+package com.cout970.magneticraft.api.internal.registries.machines.tablesieve
```
same, change package

cout970 commented on src/main/kotlin/com/cout970/magneticraft/api/internal/registries/machines/kiln/KilnRecipe.kt:
```diff
@@ -0,0 +1,47 @@
+package com.cout970.magneticraft.api.internal.registries.machines.hydraulicpress
```
same, update package

cout970 commented on src/main/kotlin/com/cout970/magneticraft/api/internal/registries/machines/kiln/KilnRecipe.kt:
```diff
@@ -0,0 +1,47 @@
+data class KilnRecipe(
+        private val input: ItemStack,
+        private val itemOutput: ItemStack?,
+        private val blockOutput: IBlockState?,
```
There should be 2 classes for Kiln recipes, one using items as output and other using a block as ouput

cout970 commented on src/main/kotlin/com/cout970/magneticraft/api/internal/registries/machines/kiln/KilnRecipeManager.kt:
```diff
@@ -0,0 +1,38 @@
+ * Created by cout970 on 22/08/2016.
+ */
+/**
+ * Internal class only please use MagneticraftApi.getHydraulicPressRecipeManager() instead
```
Update comment

cout970 commented on src/main/kotlin/com/cout970/magneticraft/api/registries/machines/kiln/IKilnRecipe.java:
```diff
@@ -0,0 +1,89 @@
+     *
+     * @return True if the recipe outputs an item
+     */
+    Boolean isItemRecipe();
```
this should be boolean instead of Boolean

cout970 reviewed (commented):
I see a lot of classes that I didn't made and have 'Created by cout970' at the start

YurgenJurgensen committed:
Minor fixes

YurgenJurgensen committed:
Minor namespace fixes

cout970 committed:
Added BlockWithStates and IStatesEnum for automatic blockstate handling
Updated BlockElectricHeater with the new utility

cout970 committed:
Merge branch 'heat' of https://github.com/Magneticraft-Team/Magneticraft into heat

cout970 committed:
Fixed render for sifter, grinder and kiln
Removed debug logging in DeviceNetworkCard
Minor fixes in class names
Minor fixes in lang files
